### PR TITLE
Clients branch

### PR DESCRIPTION
**Pull request by AmirHoseinRahmati23**
Add BaseAddress To Client

### CODE REVIEW

File: MisProject/Client/Program.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@
 builder.Services.AddTransient<CustomAuthorizationHandler>();
 builder.Services.AddHttpClient<IUserCaller, UserCaller>("Client", c =>
 {
-    c.BaseAddress = new Uri(builder.HostEnvironment.BaseAddress);
+    c.BaseAddress = "/Api/V1/User";
```
Comment (Drit-team) [inline MisProject/Client/Program.cs:44]:
```suggestion
    c.BaseAddress = new Uri(builder.HostEnvironment.BaseAddress+"/Api/V1/User");
```

File: MisProject/ClientLibraries/HttpCallers/UserCaller.cs [hunk 2]
```diff
@@ -41,7 +40,7 @@ public async Task<HttpResponseMessage> Register(RegisterDto dto)
     {
         try
         {
-            var result = await _client.PostAsJsonAsync(_controllerAddress, dto);
+            var result = await _client.PostAsJsonAsync( _client.BaseAddress , dto);
```
Comment (Drit-team) [inline MisProject/ClientLibraries/HttpCallers/UserCaller.cs:43]:
```suggestion
            var result = await _client.PostAsJsonAsync( "", dto);
```

File: MisProject/ClientLibraries/HttpCallers/UserCaller.cs [hunk 3]
```diff
@@ -55,7 +54,7 @@ public async Task<HttpResponseMessage> Register(RegisterDto dto)
     {
         try
         {
-            var result = await _client.GetAsync($"{_controllerAddress}?jwtToken={jwtToken}");
+            var result = await _client.GetAsync($"{_client.BaseAddress}?jwtToken={jwtToken}");
```
Comment (Drit-team) [inline MisProject/ClientLibraries/HttpCallers/UserCaller.cs:57]:
```suggestion
            var result = await _client.GetAsync($"?jwtToken={jwtToken}");
```

File: MisProject/ClientLibraries/HttpCallers/UserCaller.cs [hunk 4]
```diff
@@ -76,7 +75,7 @@ public async Task<HttpResponseMessage> Login(LoginDto dto)
     {
         try
         {
-            var result = await _client.PostAsJsonAsync(_controllerAddress + "/Login", dto);
+            var result = await _client.PostAsJsonAsync(_client.BaseAddress + "/Login", dto);
```
Comment (Drit-team) [inline MisProject/ClientLibraries/HttpCallers/UserCaller.cs:78]:
```suggestion
            var result = await _client.PostAsJsonAsync("/Login", dto);
```

**Review by Drit-team — CHANGES_REQUESTED** [review]

**Comment by AmirHoseinRahmati23** [comment]
test

**Pull request closed without merging**